catalog/core/catalog-core-definitionparser/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -28,7 +28,7 @@
             <reference-list interface="ddf.catalog.data.MetacardType" />
         </argument>
         <argument>
-            <reference-list interface="ddf.catalog.validation.AttributeValidator" />
+            <reference-list interface="ddf.catalog.validation.AttributeValidator" availability="optional" />
```

Comment:
Does this change fix the issues which this PR's build has had?

Reply:
yes, after fixing the first build issue, I ran into this one. So this was during DDF startup. The `catalog-core-definitionparser` bundle would now start causing DDF to have start-up issues. I made this optional since there are currently no bundles exporting `ddf.catalog.validation.AttributeValidator` in DDF. Seems to resolve this start issue and provide flexibility downstream.

Reply:
Which issue do you mean by "the first build issue"? In case I encounter one of these issues in the future, I should learn what broke, why it broke, and how you solved it.

Reply:
The first build issues was the MatchAnyValidatorTest.testMatchAnyValidatorPassesWhenValidatorListIsNull was failing with a NullPointerException.

Reply:
Oh, okay. Thanks. Do you think that your fix of the first issue caused the second issue?

Reply:
The fixes are somewhat un-related other than being validator-related. With or without the first fix, the second issue would still be there because it has a required dependency that does not exist in DDF. I only really found it because I couldn't get my PR build to pass after fixing the first issue.

Reply:
That makes sense. So between the last successful build of `2.19.x` and the creation of this PR, someone merged the removal of a required dependency?

Reply:
It was more like a required dependency/argument was added which is never found.

Reply:
Ah. Weird. Thanks!